airbyte-integrations/connectors/source-freshdesk/source_freshdesk/source.py

```diff
@@ -2,11 +2,51 @@
 # Copyright (c) 2021 Airbyte, Inc., all rights reserved.
 #
 
+import logging
+import requests
+from typing import Any, List, Mapping, Optional, Tuple
+from requests.auth import AuthBase, HTTPBasicAuth
 
-from airbyte_cdk.sources.deprecated.base_source import BaseSource
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from source_freshdesk.streams import Agents, Companies, Contacts, Conversations, Groups, Roles, SatisfactionRatings, Skills, Tickets, TimeEntries
 
-from .client import Client
 
+class SourceFreshdesk(AbstractSource):
 
-class SourceFreshdesk(BaseSource):
-    client_class = Client
+    def _create_authenticator(self, api_key: str) -> AuthBase:
+        return HTTPBasicAuth(username=api_key, password="unused_with_api_key")
+
+    def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) -> Tuple[bool, Optional[Any]]:
+        alive = True
+        error_msg = None
+        try:
+            url = f"https://{config['domain'].rstrip('/')}/api/v2/settings/helpdesk"
+            r = requests.get(url=url, auth=self._create_authenticator(config["api_key"]))
+            if not r.ok:
```

Comment:
I would suggest leveraging `r.raise_for_status()` and handling the error if it's raised.

Reply:
Sure. Updated in [24d1581](https://github.com/airbytehq/airbyte/pull/12334/commits/24d158184627496eabeb1cd9458dbc65f32a5f11).